waddrmgr/db.go

```diff
@@ -1266,10 +1303,102 @@ func upgradeManager(namespace walletdb.Namespace) error {
 		return managerError(ErrDatabase, str, err)
 	}
 
-	// Upgrade the manager as needed.
-	if version < LatestMgrVersion {
-		// No upgrades yet.
+	return nil
+}
+
+// upgradeManager upgrades the data in the provided manager namespace to newer
+// versions as neeeded.
+func upgradeManager(namespace walletdb.Namespace) error {
+	// Get the current version.
+	var version uint32
+	err := namespace.View(func(tx walletdb.Tx) error {
+		mainBucket := tx.RootBucket().Bucket(mainBucketName)
+		verBytes := mainBucket.Get(mgrVersionName)
+		version = binary.LittleEndian.Uint32(verBytes)
+		return nil
+	})
+	if err != nil {
+		str := "failed to fetch version for update"
+		return managerError(ErrDatabase, str, err)
 	}
 
+	// NOTE: There are currently no upgrades, but this is provided here as a
+	// template for how to properly do upgrades.  Each function to upgrade
+	// to the next version must include serializing the new version as a
+	// part of the same transaction so any failures in upgrades to later
+	// versions won't leave the database in an inconsistent state.  The
+	// putManagerVersion function provides a convenient mechanism for that
+	// purpose.
+	//
+	// Upgrade one version at a time so it is possible to upgrade across
+	// an aribtary number of versions without needing to write a bunch of
+	// additional code to go directly from version X to Y.
+	// if version < 2 {
+	// 	// Upgrade from version 1 to 2.
+	//	if err := upgradeToVersion2(namespace); err != nil {
+	//		return err
+	//	}
+	//
+	//	// The manager is now at version 2.
+	//	version = 2
+	// }
+	// if version < 3 {
+	// 	// Upgrade from version 2 to 3.
+	//	if err := upgradeToVersion3(namespace); err != nil {
+	//		return err
+	//	}
+	//
+	//	// The manager is now at version 3.
+	//	version = 3
+	// }
+
+	if version < 2 {
+		// The manager is now at version 2.
+		version = 2
```

Comment:
I'd move this after the call to `upgradeToVersion2` as the sample shows.  No point in changing it only to change it back imo.

Also, the comment about there currently being no upgrades should be modified since it's no longer true.